ci/ci/github.py

```diff
@@ -455,49 +473,72 @@ async def assign_gh_reviewer_if_requested(self, gh_client):
                 log.exception(f'{self.short_str()}: Unexpected exception in post to github: {data}')
 
     async def _update_github(self, gh):
-        await self._update_last_known_github_status(gh)
-        await self._update_github_review_state(gh)
-
-    @staticmethod
-    def _hail_github_status_from_statuses(statuses_json) -> Dict[str, GithubStatus]:
-        statuses = statuses_json["statuses"]
-        hail_statuses = {}
-        for s in statuses:
-            context = s['context']
-            if context == 'ci-test' or context.startswith('hail-ci'):
-                if context in hail_statuses:
-                    raise ValueError(
-                        f'github sent multiple status summaries for context {context}: {s}\n\n{statuses_json}'
-                    )
-                hail_statuses[context] = GithubStatus(s['state'])
-        return hail_statuses
-
-    async def _update_last_known_github_status(self, gh):
-        if self.source_sha:
-            source_sha_json = await gh.getitem(
-                f'/repos/{self.target_branch.branch.repo.short_str()}/commits/{self.source_sha}/status'
-            )
-            last_known_github_status = PR._hail_github_status_from_statuses(source_sha_json)
-            if last_known_github_status != self.last_known_github_status:
-                log.info(
-                    f'{self.short_str()}: new github statuses: {self.last_known_github_status} => {last_known_github_status}'
-                )
-                self.last_known_github_status = last_known_github_status
-                self.target_branch.state_changed = True
-
-    async def _update_github_review_state(self, gh):
-        review_state_query = f"""
-        query {{
-            repository(owner: "{self.target_branch.branch.repo.owner}", name: "{self.target_branch.branch.repo.name}") {{
-                pullRequest(number: {self.number}) {{
-                      reviewDecision
+        results = []
+        cursor = None
+        review_decision = -1
```

Comment:
I find using `-1` as a sentinel confusing. What does this represent? Is there a better value that could be used here?

Reply:
Very fair! I was initially using `None` to indicate that we haven't received a `review_status` from the GitHub API yet, but it turns out if the repository is not configured to require reviews, the value we receive for it from the API is also `None`. Rewrote it to represent not having received any status from the API as `None`, and having received a status but it was `None` on the API side as `"API_NONE"`.